compiler/modules/front/stmt.bal

```diff
@@ -268,17 +268,16 @@ function codeGenScope(StmtContext cx, bir:BasicBlock bb, Environment initialEnv,
         applyEffect(env, narrowings, effect);
     }
     else {
-        int lastStmtIndex = scope.stmts.length() - 1;
-        int stmtIndex = 0;
+        StmtEffect? previousEffect = ();
         foreach var stmt in scope.stmts {
+            // add narrowing to previous stmt
+            if previousEffect != () {
+                addNarrowings(cx, <bir:BasicBlock>previousEffect.block, env, previousEffect.narrowings, stmt.startPos);
```

Comment:
Why is there such a big change just to fix the position passed into addNarrowings?  Can't you just keep track of the previous stmt and use its endPos?

Reply:
Isn't simpler this way since we got rid of `lastStmtIndex` and `stmtIndex`.

Reply:
OK.